lib/new_relic/agent/instrumentation/redis/instrumentation.rb

```diff
@@ -6,38 +6,58 @@ module NewRelic::Agent::Instrumentation
   module Redis
     PRODUCT_NAME = 'Redis'
     CONNECT = 'connect'
-    UNKNOWN = "unknown"
-    LOCALHOST = "localhost"
+    UNKNOWN = 'unknown'
+    LOCALHOST = 'localhost'
     MULTI_OPERATION = 'multi'
     PIPELINE_OPERATION = 'pipeline'
+    REDIS_5 = Gem::Version.new(::Redis::VERSION) >= Gem::Version.new('5.0.0') && !!defined?(::RedisClient)
 
+    def connect_with_tracing
+      with_tracing(CONNECT, database: db) { yield }
```

Comment:
Does the monkeypatched class itself define `db`?

Reply:
Correct! It is available in all cases except the middleware, where `client.db` is defined. You can see an example of this call within #call_pipelined_with_tracing. If you'd prefer it be more specific, I could apply the `#_nr_client` method in all cases where `db` is called.